api/client.go

```diff
@@ -1143,12 +1172,20 @@ func (c *Client) rawRequestWithContext(ctx context.Context, r *Request) (*Respon
 	checkRetry := c.config.CheckRetry
 	backoff := c.config.Backoff
 	httpClient := c.config.HttpClient
+	ns := c.headers.Get(consts.NamespaceHeaderName)
 	outputCurlString := c.config.OutputCurlString
 	logger := c.config.Logger
 	c.config.modifyLock.RUnlock()
 
 	c.modifyLock.RUnlock()
 
+	switch ns {
```

Comment:
I don't understand why this change should be needed, can you explain please?

Reply:
Absolutely. When I was writing the tests I came across this pattern:
```go
	_, err = client.WithNamespace("ns1").rawRequestWithContext(
		context.Background(),
		client.NewRequest(http.MethodGet, "/"))
```
When we call `client.NewRequest()` the request will get the headers set to whatever is in `client` ([this code](https://github.com/hashicorp/vault/blob/main/api/client.go#L1101) I believe). This change will make sure at the time of the call the request has the appropriate header map.